src/types/engine_time.rs

```diff
@@ -0,0 +1,175 @@
+use std::time::Duration;
+
+use crate::raw::root::RED4ext as red;
+
+#[derive(Default, Clone, Copy)]
+#[repr(transparent)]
+pub struct EngineTime(red::EngineTime);
+
+impl EngineTime {
+    pub fn is_valid(&self) -> bool {
+        self.0.unk00.ne(&[0; 8])
+    }
+
+    pub fn as_secs_f64(&self) -> f64 {
+        f64::from_ne_bytes(self.0.unk00)
+    }
+
+    pub fn saturating_add_assign(&mut self, value: impl Into<f64>) {
+        let current = self.as_secs_f64();
+        let value: f64 = value.into();
+        let addition = current + value;
+        if addition.is_infinite() {
+            if addition.is_sign_positive() {
+                self.0.unk00 = f64::MAX.to_ne_bytes();
+            } else {
+                self.0.unk00 = f64::MIN.to_ne_bytes();
+            }
+        } else {
+            self.0.unk00 = addition.to_ne_bytes();
+        }
+    }
+
+    pub fn saturating_add(self, value: impl Into<f64>) -> Self {
+        let mut copy = self;
+        copy.saturating_add_assign(value.into());
+        copy
+    }
+
+    pub fn saturating_sub_assign(&mut self, value: impl Into<f64>) {
+        let current = self.as_secs_f64();
+        let value: f64 = value.into();
+        let addition = current - value;
+        if addition.is_infinite() {
```

Comment:
```suggestion
        if addition.is_infinite() {
```
isn't adding infinity to game time always an indicator of an error? I think maybe this could just panic

Reply:
ok, I also included `f64::NAN` in ae5c0a9.